tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigProviderTest.java

```diff
@@ -71,8 +71,14 @@ public static WebArchive deploy() {
     @Test
     public void testEnvironmentConfigSource() {
         Map<String, String> env = System.getenv();
+        Properties properties = System.getProperties();
         for (Map.Entry<String, String> envEntry : env.entrySet()) {
-            Assert.assertEquals(envEntry.getValue(), config.getValue(envEntry.getKey(), String.class));
+            String key = envEntry.getKey();
+            if(!properties.containsKey(key)) {
+                String value = envEntry.getValue();
+                String cfgValue = config.getValue(key, String.class);
+                Assert.assertEquals(value, cfgValue);
```

Comment:
Need to verify whether this line is actually hit, so that we tested something.

Reply:
What about testing that a property DOES override the env value if present - which would match what the impl has to do more closely?

Reply:
That test will be to test system property override environment variables. It is nice to have but I don't know whether it is reliable or not, as I think in a lot of environments, the variables are set to the same value.